bpf/lib/encap.h

```diff
@@ -237,6 +240,12 @@ encap_and_redirect_lxc(struct __ctx_buff *ctx, __u32 tunnel_endpoint,
 	if (!tunnel)
 		return DROP_NO_TUNNEL_ENDPOINT;
 
+#if defined(TUNNEL_MODE) && defined(ENCRYPTION_STRICT_MODE)
+	if (!strict_allow(ctx))
+		return send_drop_notify_error(ctx, 0, DROP_UNENCRYPTED_TRAFFIC,
+					      CTX_ACT_DROP, METRIC_EGRESS);
+#endif
```

Comment:
Why is this check placed here? We may still encrypt the packet below and we may return without encryption just above. I think it would be a lot safer to run this check in bpf_overlay instead. That way we ensure that we haven't missed a case: if a pod-to-pod packet reaches bpf_overlay unencrypted, then it should be dropped.

Reply:
The check is/was here because I tried the place that were obvious to me, but those didn't work. So I tried the less obvious ones until the test passed. Do you have any specific point in mind? moving this to bpf_overlay:542 or bpf_overlay:203 doesn't seem to work.

Reply:
The exact same check in bpf_overlay:542 should work. I'm happy to help debug this over a Slack thread if it still doesn't work.

Reply:
Thanks a lot for the discussion and help! I think the current proposal should resolve this now.